### PR TITLE
[linux] handle stop events while stepping

### PR DESCRIPTION
**Pull request by andrurogerz**
## Purpose
Properly handle a case that occasionally triggers [this](https://github.com/compnerd/ds2/blob/main/Sources/Target/Linux/Process.cpp#L274) `DS2BUG` fatal error in `Linux::Process::wait` when running some multi-threaded lldb tests.

## Overview
* Handle the case where we get a stop event for a single-stepping thread with `kEventNone` and `kReasonNone`
* Enable 15 lldb test cases that were previously flaky due to this bug
* Rework the `Linux::Process::wait` function be a little bit more readable
* Handle all failure cases in `Linux::Process::wait` with the `CHK()` macro

## Problem Details
The scenario where a `kEventNone` event with a reason other than `kReasonThreadSpawn` is alluded to in [this](https://github.com/compnerd/ds2/blob/main/Sources/Target/Linux/Process.cpp#L236-L238) comment block of `Linux::Process::wait` but it suggests it will never happen while the thread is single stepping. However, this scenario can legitimately occur and is fairly easy to reproduce with the lldb multithreaded tests.  In this case, the reason will be `kReasonNone`.

The implementation already handled this `kEventNone` and `kReasonNone` event by continuing the thread as long as it wasn't also being single-stepped. This change updates it to step the thread in the same situation if it is being single-stepped. However, I haven't 100% convinced myself the implementation was correct to begin with (it could probably use an overhaul). Intuitively, I would expect to ignore events with `kEventNone` and `kReasonNone`, but that's not what it was doing previously.

## Validation
The re-enabled tests now pass consistently locally on Android and Linux, and have now passed on at least 10 automated runs.

Manually set some breakpoints and stepped in a simple multi-threaded program. It behaved as expected.

### CODE REVIEW

File: Sources/Target/Linux/Process.cpp [hunk 1]
```diff
@@ -232,48 +232,54 @@ ErrorCode Process::wait() {
...
+        if (stepping)
+          CHK(_currentThread->step());
+        else
+          CHK(_currentThread->resume());
+        break;
+
+      case StopInfo::kReasonThreadSpawn:
+        if (!stepping) // (2a)
+          CHK(_currentThread->resume());
+        else { // (2b)
```
Comment (compnerd) [inline Sources/Target/Linux/Process.cpp:247]:
```suggestion
        if (!stepping) { // (2a)
          CHK(_currentThread->resume());
        } else { // (2b)
```

Comment (compnerd) [reply Sources/Target/Linux/Process.cpp:247]: Keeping the braces on both sides helps readability.

File: Sources/Target/Linux/Process.cpp [hunk 2]
```diff
@@ -232,48 +232,54 @@ ErrorCode Process::wait() {
...
+          unsigned long spawnedThreadIdData;
+          CHK(ptrace().getEventMessage(_pid, spawnedThreadIdData));
+          auto const spawnedThreadId = static_cast<ThreadId>(spawnedThreadIdData);
+          if (_threads.find(spawnedThreadId) == _threads.end()) {
+            // If the newly cloned thread does not yet exist within the _threads
+            // vector then we have not yet seen it with waitpid(2). The most
+            // recent
+            // waitpid(2) returns a status guaranteeing a clone event did happen.
+            // Therefore, we wait and block until we see the cloned thread.
```
Comment (compnerd) [inline Sources/Target/Linux/Process.cpp:256]:
```suggestion
            // If the newly cloned thread does not yet exist within the _threads
            // vector then we have not yet seen it with waitpid(2). The most
            // recent waitpid(2) returns a status guaranteeing a clone event did happen.
            // Therefore, we wait and block until we see the cloned thread.
```

File: Sources/Target/Linux/Process.cpp [hunk 3]
```diff
@@ -232,48 +232,54 @@ ErrorCode Process::wait() {
...
+            int spawnedThreadStatus;
+            ThreadId returnedThreadId =
+                blocking_waitpid(spawnedThreadId, &spawnedThreadStatus, __WALL);
+            if (spawnedThreadId != returnedThreadId)
+              return kErrorProcessNotFound;
+            DS2LOG(Debug, "child thread tid %d %s", returnedThreadId,
+                  Stringify::WaitStatus(spawnedThreadStatus));
```
Comment (compnerd) [inline Sources/Target/Linux/Process.cpp:263]: Can't tell if GH or the indentation is off.

**Review by compnerd — APPROVED** [review]

File: Sources/Target/Linux/Process.cpp (hunk 3 above)
Comment (andrurogerz) [reply Sources/Target/Linux/Process.cpp:262]: Indentation is off by 1 space; fixing.

**Pull request merged**